lib/signs_ui_web/router.ex

```diff
@@ -68,4 +64,18 @@ defmodule SignsUiWeb.Router do
         |> halt()
     end
   end
+
+  @spec cognito_auth(Plug.Conn.t(), Plug.opts()) :: Plug.Conn.t()
+  defp cognito_auth(conn, _) do
+    conn = fetch_session(conn)
+    login_expiration = get_session(conn, :login_expiration)
+
+    current_time = DateTime.utc_now() |> DateTime.to_unix()
+
+    if is_nil(login_expiration) or current_time >= login_expiration do
+      conn |> redirect(to: "/cognito/new") |> halt()
```

Comment:
I think `/cognito/new` has a path helper function. Try `mix phx.routes` to see, and then I think you might need to do something like `Router.foo_path()` to access it. Although, come to think of it you might not be able to from _within_ the router...

Reply:
Hmm, so I'm trying `Router.cognito_path` and not getting any luck. I've tried it both without arguments, and also with `:new`. It's worth noting that `mix phx.routes` reports two different entries for `cognito_path` (one for `/cognito` and one for `/cognito/new`), but both `cognito_path/0` and `cognito_path/1` are apparently undefined.

Reply:
The path helper is typically at least `/2` (or more if takes an ID). The first argument is either a `conn` if you have one, or the module name for the Endpoint. So try `SignsUiWeb.Router.Helpers.cognito_path(conn, :new)`.

Reply:
Ah, thanks. I managed to get it working.